Bumps the mkdocs opal version

fredkingham opened the pull request:
Bumps the mkdocs version to 0.24.0

Commit:
Bumps the mkdocs opal version

davidmiller commented:
Can we confirm that the documentation definitely still compiles with the new version

Commit:
Pins Markdown version to <3.4

This is because Markdown 3.4+ raises an error with extendMarkdown() missing 1 required positional argument: 'md_globals'.

This is because the version broke some plugins including the one we use.
